rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+Get learners to reflect on what makes it easy to navigate an unfamiliar project.  
+
+TODO: Roadmap for the chapter: 
+
+* What is a project? What should every project have?
+* How should I structure the directories and files?
+* How should I document data? 
+
+## What *is* a project? {#project-thinking}
+
+This chapter assumes each project is a separate Git repository, 
+and "project structure" describes the names, locations (and sometimes content) of the directories and files in this repository.  
+But, how do you know what should be considered a project?
+
+[Good enough practices in scientific computing](https://doi.org/10.1371/journal.pcbi.1005510):
+
+> "Like deciding when a chunk of code should be made a function, the ultimate goal of dividing research into distinct projects is to help you and others best understand your work. Some researchers create a separate project for each manuscript they are working on, while others group all research on a common theme, data set, or algorithm into a single project."
+
+> "As a rule of thumb, divide work into projects based on the overlap in data and code files. If 2 research efforts share no data or code, they will probably be easiest to manage independently. If they share more than half of their data and code, they are probably best managed together, while if you are building tools that are used in several projects, the common code should probably be in a project of its own." 
+
+Like features (Chapter \@ref(branches)),
+what exactly constitutes a "project" requires a bit of judgment,
+and different people will make different decisions.
+Some common criteria are one project per publication,
+one project per deliverable piece of software,
+or one project per team.
+The first tends to be too small:
+a good data set will result in several reports,
+and the goal of some projects is to produce a steady stream of reports (such as monthly forecasts).
+The second is a good fit for software engineering projects
+whose primary aim is to produce tools rather than results,
+but is inappropriate for most data analysis work.
+The third tends to be too large:
+a team of half a dozen people may work on many different things at once,
+and a repository that holds them all quickly looks like someone's basement.
+
+The best rule of thumb we have found for deciding what is and isn't a project
+is to ask what you have meetings about.
+If the same set of people need to get together on a regular basis to talk about something,
+that "something" probably deserves its own repository.
+And if the list of people changes slowly over time but the meetings continue,
+that's an even stronger sign.
+
+## What files should every project contain? {#project-boilerplate}
+
+TODO: What are the bare minimums here?  I'd say must have `README`, even for your own projects that only live locally.  As soon as something is on GitHub must have `LICENSE` and `CONDUCT`?
```

Comment:
Agree with readme for local and adding license and conduct when public on github (maybe license as soon as it is on github and conduct when/if you want to have contributors?).